### DISCUSSION TITLE
Logitech Options+

### DISCUSSION THREAD

**Discussion by LunaticMuch** [Homebrew/discussions#2492] [Casks]
I am seeing Logitech Options+ is now a thing, although still branded as a beta. Is there any way to get it via brew? 
I inspected a bit the [code of the current cask](https://github.com/Homebrew/homebrew-cask-drivers/blob/master/Casks/logitech-options.rb) but I do not find the link for Options+ in the JSON files the formula gets from Logitech.

I would be happy to help even writing the formula if I can find how to get the download URL.

**Comment by neutric** [comment]
I might be wrong, but don't the [Beta Product Terms](https://www.logitech.com/en-us/legal/beta-test-agreement.html) from [this](https://www.logitech.com/en-us/software/logi-options-plus.html#join-beta) prohibit Logitech Options+ Beta from being installed via Homebrew? And then there is the matter of [Acceptable Casks](https://docs.brew.sh/Acceptable-Casks#finding-a-home-for-your-cask), more specifically the "Walled" part of it.

**Reply by LunaticMuch** [reply]
I did not consider it. Need to go through the terms. 
Actually, I came across the idea of a formula because their installer is crappy and had a lot of proxy issues. Reported it, twice, but it was like sending to `/dev/null`.

**Reply by carlocab** [reply]
Formulae must be open source, so this can't be a formula. Casks can be closed-source, but not all closed-source software can be casks.

**Comment by kmcnellis** [comment]
Looks like it's no longer in beta, and the download link is freely available (https://download01.logi.com/web/ftp/pub/techsupport/optionsplus/logioptionsplus_installer.zip)

Maybe this can be reconsidered?

**Reply by p-linnane** [reply]
If you'd like to open a PR we will be happy to review it.

**Reply by chrishuan9** [reply]
I briefly looked into the Logitech Options+ Installer. Basically the Installer doesn't come with all the files and during the installation it downloads even more files (indicated in the setup installer ~500MB's). This will probably be needed to be considered in the formula.

**Reply by colinhuckstep** [reply]
There is an offline installer available here. Though I've noticed it sometimes lags behind the latest version.

https://download01.logi.com/web/ftp/pub/techsupport/optionsplus/logioptionsplus_installer_offline.zip

**Reply by pattyland** [reply]
Should `Logi Options+` be treated as an Update or as a new cask? There might be reasons people would like to stay on `Logitech Options`?

**Reply by colinhuckstep** [reply]
It should be a separate cask. Not all devices are currently supported in Options+. They've been adding devices over time.

**Comment by ZhongRuoyu** [comment]
`logi-options-plus` was previously in https://github.com/Homebrew/homebrew-cask-drivers. Now it has been migrated to https://github.com/Homebrew/homebrew-cask (in https://github.com/Homebrew/homebrew-cask/pull/145059), so you should be able to do `brew install --cask logi-options-plus` without having to tap `Homebrew/cask-drivers`.

**Comment by iamtodor** [comment]
Was trying to install it today by `brew install --cask logi-options-plus` and seems like it failed, cause the app didn't open correctly.
MacOS 13.6.1 (22G313), intel based

**Comment by architchandra** [comment]
Came here looking for the correct cask. Just in case it helps anyone else, the cask with the name `logi-options-plus` has now been renamed to `logi-options+`.

Here's the output from brew info `logi-options-plus`:

```
❯ brew info logi-options-plus
Warning: Cask logi-options-plus was renamed to logi-options+.
==> logi-options+: 1.82.618412 (auto_updates)
https://www.logitech.com/en-us/software/logi-options-plus.html
Installed
/opt/homebrew/Caskroom/logi-options+/1.38.397153 (42.5MB)
From: https://github.com/Homebrew/homebrew-cask/blob/HEAD/Casks/l/logi-options+.rb
==> Name
Logitech Options+
==> Description
Software for Logitech devices
==> Artifacts
logioptionsplus_installer.app/Contents/MacOS/logioptionsplus_installer (Installer)
==> Caveats
You must reboot for the installation of logi-options+ to take effect.

==> Analytics
install: 2,048 (30 days), 3,582 (90 days), 3,583 (365 days)
```
